src/inference/InterfaceExtensions.jl

```diff
@@ -0,0 +1,452 @@
+module InterfaceExtensions
+
+using ..Turing
+using ..Interface
+import MCMCChains: Chains
+
+import ..Interface: AbstractTransition, sample, step!, sample_init!,
+    transitions_init, sample_end!, AbstractSampler, transition_type,
+    callback, init_callback, AbstractCallback
+import ..Turing: Model, Sampler, PROGRESS
+import ..RandomVariables: islinked, invlink!, getlogp, tonamedtuple
+import ..Inference: InferenceAlgorithm, ParticleInference, AHMC, Hamiltonian,
+                    StaticHamiltonian, AdaptiveHamiltonian
+import ProgressMeter
+
+export sample,
+       sample_init!,
+       sample_end!,
+       set_resume!,
+       resume,
+       save,
+       transition_type,
+       transitions_init,
+       transition,
+       Transition,
+       ParticleTransition,
+       Chains,
+       initialize_parameters!
+
+# Internal variable names for MCMCChains.
+const TURING_INTERNAL_VARS =
+    (internals = ["elapsed", "eval_num", "lf_eps", "lp", "weight", "le",
+                  "acceptance_rate", "hamiltonian_energy", "is_accept", 
+                  "log_density", "n_steps", "numerical_error", "step_size",
+                  "tree_depth"],)
+
+####################
+# Transition Types #
+####################
+
+######################
+# Default Transition #
+######################
+
+struct Transition{T} <: AbstractTransition
+    θ  :: T
+    lp :: Float64
+end
+
+function transition(spl::Sampler)
+    theta = tonamedtuple(spl.state.vi)
+    lp = getlogp(spl.state.vi)
+    return Transition{typeof(theta)}(theta, lp)
+end
+
+function transition(spl::Sampler, nt::NamedTuple)
```

Comment:
Perhaps consider unifying the following two functions 

- `transition(spl::Sampler)`
- `transition(spl::Sampler, nt::NamedTuple)`

into one function with default value for `nt`

- `transition(spl::Sampler, nt::NamedTuple=NamedTuple())`